ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -25,103 +25,301 @@ desc: Handles callbacks and varargs
 #include <utility>
 
 #include "ldr_ptrs.inl"
+#ifdef _M_ARM_64
+__attribute__((naked))
+void* InternalVariadic(void* Ptr, uint64_t a_0, size_t count, void** list) {
+  // Arguments 0-7 is in registers
```

Comment:
There's no guarantee that they are stored in integer registers, fp arguments are stored in fp registers, in a way you can't parse them without knowing the argument order and count

Reply:
printf("a", 1, 1.0, 2, 2.0) ends up putting
x0 = "a"
x1 = 1
x2 = 2
fp0=1.0
fp1 = 2.0

Reply:
This isn't meant to be completely generic to all variadic packing packing. I'll rename it to make that more clear that it is just for flat uint64_t